homeassistant/components/rflink/light.py

```diff
@@ -188,6 +189,20 @@ async def async_turn_on(self, **kwargs):
         # Turn on light at the requested dim level
         await self._async_handle_command("dim", self._brightness)
 
+    def _handle_event(self, event):
+        """Adjust state if Rflink picks up a remote command for this device."""
+        self.cancel_queued_send_commands()
+
+        command = event["command"]
+        if command in ["on", "allon"]:
+            self._state = True
+        elif command in ["off", "alloff"]:
+            self._state = False
+        # dimmable device accept 'set_level=(0-15)' commands
+        elif re.search("^set_level=(0?[0-9]|1[0-5])$", command, re.IGNORECASE):
+            self._brightness = int(command.split("=")[1]) * 17
```

Comment:
Almost the same code is also in two places in `rflink/light.py`, consider moving it to a helper function in new file `rflink/utils.py` to make it easier to maintain.

Reply:
The hierarchy of the classes has been modified and I believe that (within the type `light` entities) there is no code that can be externalized.
If you can tell me what you are referring to, I will check it without problem.

Reply:
Sure, I meant this line, which converts rflink brightness 0..15 to 0..255:
```py
self._brightness = int(command.split("=")[1]) * 17
```

There's something similar happening in two other places, where the idea is to force the brightness to one of the 16 steps supported by rflink:
```py
self._brightness = int(kwargs[ATTR_BRIGHTNESS] / 17) * 17
```

In both cases the code is quite unclear with the number 17 appearing with no explanation.
Maybe this could be refactored to a (commented) helper function instead?